Lib/test/test_fstring.py

```diff
@@ -427,703 +482,1103 @@ def test_ast_line_numbers_with_parentheses(self):
+            ],
+        )
 
     def test_compile_time_concat(self):
-        x = 'def'
-        self.assertEqual('abc' f'## {x}ghi', 'abc## defghi')
-        self.assertEqual('abc' f'{x}' 'ghi', 'abcdefghi')
-        self.assertEqual('abc' f'{x}' 'gh' f'i{x:4}', 'abcdefghidef ')
-        self.assertEqual('{x}' f'{x}', '{x}def')
-        self.assertEqual('{x' f'{x}', '{xdef')
-        self.assertEqual('{x}' f'{x}', '{x}def')
-        self.assertEqual('{{x}}' f'{x}', '{{x}}def')
-        self.assertEqual('{{x' f'{x}', '{{xdef')
-        self.assertEqual('x}}' f'{x}', 'x}}def')
-        self.assertEqual(f'{x}' 'x}}', 'defx}}')
-        self.assertEqual(f'{x}' '', 'def')
-        self.assertEqual('' f'{x}' '', 'def')
-        self.assertEqual('' f'{x}', 'def')
-        self.assertEqual(f'{x}' '2', 'def2')
-        self.assertEqual('1' f'{x}' '2', '1def2')
-        self.assertEqual('1' f'{x}', '1def')
-        self.assertEqual(f'{x}' f'-{x}', 'def-def')
-        self.assertEqual('' f'', '')
-        self.assertEqual('' f'' '', '')
-        self.assertEqual('' f'' '' f'', '')
-        self.assertEqual(f'', '')
-        self.assertEqual(f'' '', '')
-        self.assertEqual(f'' '' f'', '')
-        self.assertEqual(f'' '' f'' '', '')
-
-        self.assertAllRaise(SyntaxError, "f-string: expecting '}'",
-                            ["f'{3' f'}'",  # can't concat to get a valid f-string
-                             ])
+        x = "def"
+        self.assertEqual("abc" f"## {x}ghi", "abc## defghi")
+        self.assertEqual("abc" f"{x}" "ghi", "abcdefghi")
+        self.assertEqual("abc" f"{x}" "gh" f"i{x:4}", "abcdefghidef ")
+        self.assertEqual("{x}" f"{x}", "{x}def")
+        self.assertEqual("{x" f"{x}", "{xdef")
+        self.assertEqual("{x}" f"{x}", "{x}def")
+        self.assertEqual("{{x}}" f"{x}", "{{x}}def")
+        self.assertEqual("{{x" f"{x}", "{{xdef")
+        self.assertEqual("x}}" f"{x}", "x}}def")
+        self.assertEqual(f"{x}" "x}}", "defx}}")
+        self.assertEqual(f"{x}" "", "def")
+        self.assertEqual("" f"{x}" "", "def")
+        self.assertEqual("" f"{x}", "def")
+        self.assertEqual(f"{x}" "2", "def2")
+        self.assertEqual("1" f"{x}" "2", "1def2")
+        self.assertEqual("1" f"{x}", "1def")
+        self.assertEqual(f"{x}" f"-{x}", "def-def")
+        self.assertEqual("" f"", "")
+        self.assertEqual("" f"" "", "")
+        self.assertEqual("" f"" "" f"", "")
+        self.assertEqual(f"", "")
+        self.assertEqual(f"" "", "")
+        self.assertEqual(f"" "" f"", "")
+        self.assertEqual(f"" "" f"" "", "")
+
+        # This is not really [f'{'] + [f'}'] since we treat the inside
+        # of braces as a purely new context, so it is actually f'{ and
+        # then eval('  f') (a valid expression) and then }' which would
+        # constitute a valid f-string.
+        # TODO: RUSTPYTHON SyntaxError
+        # self.assertEqual(f'{' f'}', " f")
```

Comment:
I can't remove some comments, because it makes SyntaxError before test runs (so it cannot be passed by expectedFailure)

Reply:
Ah, that makes sense. Thank you